.drone/drone.yml

```diff
@@ -177,20 +170,85 @@ steps:
       - mkdir -p $HOME/.docker
       - printenv GCR_CREDS > $HOME/.docker/config.json
       - docker login -u $DOCKER_LOGIN -p $DOCKER_PASSWORD
-      - apk update && apk add make bash wget git qemu
-      - wget -q https://github.com/docker/buildx/releases/download/v0.5.1/buildx-v0.5.1.linux-amd64
-      - mkdir -p ~/.docker/cli-plugins
-      - cp buildx-v0.5.1.linux-amd64 ~/.docker/cli-plugins/docker-buildx
-      - chmod a+x ~/.docker/cli-plugins/docker-buildx
+
+      # Create a buildx worker container for multiplatform builds.
       - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       - docker buildx create --name multiarch --driver docker-container --use
-      - export RELEASE_TAG=${DRONE_TAG}
-      - export IMAGE_BRANCH_TAG=${DRONE_BRANCH}
-      - make DOCKER_OPTS="" DRONE=true CROSS_BUILD=true BUILD_IN_CONTAINER=true RELEASE_BUILD=true agent-image
-      - make DOCKER_OPTS="" DRONE=true CROSS_BUILD=true BUILD_IN_CONTAINER=true RELEASE_BUILD=true agentctl-image
-      - make DOCKER_OPTS="" DRONE=true CROSS_BUILD=true BUILD_IN_CONTAINER=true RELEASE_BUILD=true agent-operator-image
-      - make DOCKER_OPTS="" DRONE=true CROSS_BUILD=true BUILD_IN_CONTAINER=true RELEASE_BUILD=true grafana-agent-crow-image
-      - make DOCKER_OPTS="" DRONE=true CROSS_BUILD=true BUILD_IN_CONTAINER=true RELEASE_BUILD=true agent-smoke-image
+
+      - |
+        # If containers are being built as the result of a tag, force the
+        # VERSION to come from the drone tag info.
+        #
+        # Otherwise, use the script to determine the tag.
+        if [ ! -z "$DRONE_TAG" ]; then
+          export VERISON=$DRONE_TAG
+        else
+          export VERSION=$(./tools/image-tag)
+        fi
+
+        # We also need to know which "branch tag" to update. If this is a
+        # stable release (e.g., not a release candiate, we'll update :latest,
+        # otherwise we update the branch name).
+        #
+        # If there's no branch, we'll default to the version.
+        if [ ! -z "$DRONE_TAG" ] && [[ "$DRONE_TAG" != *"-rc."* ]]; then
+          export BRANCH_TAG=$DRONE_TAG
+        elif [ ! -z "$DRONE_BRANCH" ]; then
+          export BRANCH_TAG=$DRONE_BRANCH
+        else
+          export BRANCH_TAG=$VERSION
+        fi
+
+      - >-
+        docker buildx build --push
+        --platform linux/amd64,linux/arm64,linux/arm/v7,linux/ppc64le
+        --build-arg RELEASE_BUILD=1
+        --build-arg VERISON=$VERSION
+        -t grafana/agent:$VERSION
+        -t grafana/agent:$BRANCH_TAG
+        -f cmd/agent/Dockerfile.new
+        .
```

Comment:
This is a bit wordy, but it doesn't feel appropriate to go in the Makefile since:

1. You _have_ to push the images when doing a multiplatform build, which is inappropriate for local development
2. There's a lot of Drone-specific logic around determining what the tags should be.

This is probably something we can clean up a bit by using Jsonnet instead of YAML for Drone, but that's a different PR for a different day.

Reply:
I'd be ok with a local script for this if is is kinda ugly in the drone (which it is). Might be easier to test locally in that case, even if you have to contrive some vars.

Reply:
👍 I'll create a new script for building cross-platform images.